cmd/insprd/auth/jwt/auth.go

```diff
@@ -0,0 +1,86 @@
+// Package jwtauth is responsible for implementing the auth
+// methods specified in the auth folder of the inspr pkg.
+package jwtauth
+
+import (
+	"crypto/rsa"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/lestrrat-go/jwx/jwa"
+	"github.com/lestrrat-go/jwx/jwt"
+	"gitlab.inspr.dev/inspr/core/pkg/auth/models"
+)
+
+// JWTauth structure containing the private key of in the service side,
+// this key is used to parse the user keys given in the requests.
+type JWTauth struct {
+	rsaKey *rsa.PrivateKey
+}
+
+// NewJWTauth takes an *rsa.PrivateKey and returns an
+// structure that implements the auth interface
+func NewJWTauth(privateKey *rsa.PrivateKey) *JWTauth {
+	return &JWTauth{
+		rsaKey: privateKey,
+	}
+}
+
+// Validade is a wrapper that checks the token of the http request and if it's
+// valid, proceeds to execute the request and if it isn't valid returns an error
+func (JA *JWTauth) Validade(token []byte) (models.Payload, []byte, error) {
+
+	jwtToken, err := jwt.Parse(
+		token,
+		jwt.WithValidate(true),
+		jwt.WithVerify(jwa.RS256, &JA.rsaKey.PublicKey),
+	)
+
+	// not valid token
+	if err != nil {
+		return models.Payload{}, token, err
+	}
+
+	expiration, found := jwtToken.Get(jwt.ExpirationKey)
+	now := time.Now()
+
+	// expired
+	if !found || now.After(expiration.(time.Time)) {
```

Comment:
Don't think time.After fits here

Reply:
There is a distinction between the function After in the Time packge and the function (time.Time).After.

The second one is attached to the time.Time structure which implements the comparison between the structure and the time given in the parameter.